pkg/parser/parser.go

```diff
@@ -17,6 +18,12 @@ type Parser interface {
 
 // GetParser returns a Parser for the given file type. Defaults to returning the YAML parser.
 func GetParser(fileName string) Parser {
+	if filepath.Base(fileName) == "Dockerfile" {
```

Comment:
If you rebase against latest commit, you can do this in `input.go`. This will also support input flag like, `--input dockerfile`

	if inputType != "" {
		return &Input{
			input: inputType,
			fName: fileName,
		}
	}
	if filepath.Base(fileName) == "Dockerfile" {
		suffix = "Dockerfile"
	} else {
            //also support like test.dockerfile or staging.dockerfile
		suffix := strings.Replace(filepath.Ext(fileName), ".", "", -1)
	}
	return &Input{
		input: suffix,
		fName: fileName,
	}

and the `parser.go` will like this,

	switch i.input {
	case "Dockerfile", "dockerfile":
		return &docker.Parser{
			FileName: fileName,
		}

Reply:
Yep, rebased it on master. It was quite an old branch.